### PR TITLE
Connect prototype branch switcher to Github endpoints

### PR DESCRIPTION
**Pull request by jbevis**
This PR will make use of the `list-branches` and `create-branch` endpoints in the the prototype BranchSwitcher plugin.

### CODE REVIEW

File: packages/@tinacms/cli/src/cmds/start-server/server.ts [hunk 1]
```diff
@@ -61,6 +61,39 @@ const gqlServer = async () => {
     return res.json(result)
   })
 
+  app.get('/list-branches', async (req, res) => {
+    try {
+      const { query } = req
+      const { owner, repo } = query
+      const result = await gqlPackage.listBranches({
+        auth: process.env.GITHUB_PERSONAL_ACCESS_TOKEN,
```
Comment (logan-anderson) [inline packages/@tinacms/cli/src/cmds/start-server/server.ts:69]: won't this have to be proxied though `tina-cloud?

Comment (jbevis) [reply packages/@tinacms/cli/src/cmds/start-server/server.ts]: Yes, I imagine we'll be taking steps to get this cleaned up and more in line with some best practices. This first pass is really, "Hey, can this work?"

**Comment by logan-anderson** [comment]
I pulled down the code and added this to my `.env` (in tina-cloud-starter) file
```
NEXT_PUBLIC_GITHUB_NAME=logan-anderson
NEXT_PUBLIC_GITHUB_REPO=tina-cloud-starter
NEXT_PUBLIC_GITHUB_PERSONAL_ACCESS_TOKEN=***
```

Then ran a build, when I run `tina-cloud-starter` I get this error.

```
API rate limit exceeded for 137.149.182.201. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)
Error: Unable to fetch, errors: 
        Unable to find record 'examples/tina-cloud-starter/.tina/__generated__/_graphql.json' in Github Repository: 'tinacms/tinacms', Ref: 'main'
```

**Comment by logan-anderson** [comment]
Made a PR into this one that fixes the build error @jbevis. See https://github.com/tinacms/tinacms/pull/2106

**Pull request closed without merging**